### PR TITLE
Wip improvements

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
@jmburges 

addresses issue #1 #3

### CODE REVIEW

**Comment by jmburges** [comment]
This is awesome...but I talked to @StevenNunez and we really don't believe it belongs this early in the content. Probably better to put it in the "advanced ruby" content we are creating. Any reason why that wouldn't work?

**Comment by SophieDeBenedetto** [comment]
yea i think that makes sense. so the only things in the inheritance and modules unit would be an inheritance code along and a modules code along. i'll move this.

File: README.md [hunk 1]
```diff
@@ -1,66 +1,292 @@
...
+
+First, run the test suite. Wow, we're passing all of our tests! Okay, not let go of those passing tests because we are about to break our code. 
+
+The first area of refactoring we'll be attacking are the class methods. Notice that both the `Song` and `Artist` class have `.count`, `reset_all` and `find_by_name` class methods. Instead of repeating the same exact code in both classes, let's extract these class methods into a module that we can *extend* into the classes.
+
+Ready to break your code? Comment out the `reset_all`, `count` and `find_by_name` methods in the `Song` and `Artist` class. Run your test suite. Phew! Okay, we did it. That wasn't so bad, was it?
+
+#### The `Memorable` Module
+
+Let's define our module. Create a `concerns` folder inside `lib`. This is where we'll store our modules. Inside the `concerns` folder, create a file called `memorable.rb`. Open up that file and define a module:
```
Comment (jmburges) [inline README.md]: Why are you putting this stuff in the `concerns` folder? Do students know what that folder is?

File: README.md [hunk 2]
```diff
@@ -1,66 +1,292 @@
...
+module Paramable
+end
+```
+
+Build the `to_param` method inside your module and use the `include` keyword to include the `Paramable` module in both the `Song` and `Artist` class. Once you get your tests passing again, go ahead and delete the commented-out `to_param` method from the `Song` and `Artist` classes. 
+
+
+#### Advanced: The `to_param` Method
+
+In a previous unit, we build a  `slugify` method that made a human-readable file name for each music file that we were creating and writing to. A "slug" can refer to either a file name or a URL. For example, you might visit www.facebook.com/your-name.
```
Comment (jmburges) [inline README.md]: "we built*

Comment (jmburges) [inline README.md]: lab dependency

File: README.md [hunk 3]
```diff
@@ -1,66 +1,292 @@
...
+### Bonus: Refactoring the `.initialize` Method
+
+#### Recognizing Repetition
+
+Let's take a look at the `.initialize` methods of both the `Song` and `Artist` class: 
+
+#### `lib/song.rb`
+
+```ruby
+ def .initialize
```
Comment (jmburges) [inline README.md]: the dot (`.`) there. Is that proper syntax?

File: README.md (hunk 2 above)
Comment (SophieDeBenedetto) [inline README.md]: fixed.

File: README.md (hunk 1 above)
Comment (SophieDeBenedetto) [inline README.md]: fixed

File: README.md (hunk 3 above)
Comment (SophieDeBenedetto) [inline README.md]: fixed.

**Comment by SophieDeBenedetto** [comment]
@jmburges, all comments addressed

**Comment by AnnJohn** [comment]
@SophieDeBenedetto Can you merge this one?

**Comment by SophieDeBenedetto** [comment]
I'd like to but I usually wait for joe to review my changes in light of his comments. then if he says ship it, i merge and close the issue. feel free to check in with him though if this workflow should change.

**Comment by AnnJohn** [comment]
@SophieDeBenedetto Talked to Joe. He's fine with having this one merged, and we'll talk about clarifying the workflow (not a new one) next week.  Thanks

**Pull request merged**